### PR TITLE
Add system tests for RFmxSpecAn

### PR DESCRIPTION
**Pull request by astarche**
### What does this Pull Request accomplish?

Add system level tests based on examples and exercising datatypes that require conversion.

### Why should this Pull Request be merged?

Fills out acceptance table for initial SpecAn extraction feature.

### What testing has been done?

Ran and passed all RFmx Tests.

Stepped through debugger on attribute conversion tests to ensure that the values were correct when passed to the driver.

### CODE REVIEW

File: generated/nirfmxspecan/nirfmxspecan_service.cpp [hunk 1]
```diff
@@ -11703,6 +11703,9 @@ namespace nirfmxspecan_grpc {
     catch (nidevice_grpc::LibraryLoadException& ex) {
       return ::grpc::Status(::grpc::NOT_FOUND, ex.what());
     }
+    catch (nidevice_grpc::ValueOutOfRangeException& ex) {
```
Comment (astarche) [inline generated/nirfmxspecan/nirfmxspecan_service.cpp:11706]: Note: scalar coercions like these are not actually implemented. Filed #446

**Review by gregstoll — APPROVED** [review]

File: source/tests/system/nirfmxspecan_driver_api_tests.cpp [hunk 2]
```diff
@@ -106,10 +160,153 @@ TEST_F(NiRFmxSpecAnDriverApiTests, SpectrumBasicFromExample_DataLooksReasonable)
...
+      client::set_attribute_u16(stub(), session, "", NiRFmxSpecAnAttribute::NIRFMXSPECAN_ATTRIBUTE_NF_EXTERNAL_PREAMP_PRESENT, 400));
+}
+
 TEST_F(NiRFmxSpecAnDriverApiTests, SetAndGetAttributeInt32_Succeeds)
 {
   auto session = init_session(stub(), PXI_5663);
-  EXPECT_SUCCESS(session, client::set_attribute_i32(stub(), session, "", NiRFmxSpecAnAttribute::NIRFMXSPECAN_ATTRIBUTE_NF_EXTERNAL_PREAMP_PRESENT, NiRFmxSpecAnInt32AttributeValues::NIRFMXSPECAN_INT32_NF_EXTERNAL_PREAMP_PRESENT_TRUE));
+  EXPECT_SUCCESS(
+      session,
+      client::set_attribute_i32(stub(), session, "", NiRFmxSpecAnAttribute::NIRFMXSPECAN_ATTRIBUTE_NF_EXTERNAL_PREAMP_PRESENT, NiRFmxSpecAnInt32AttributeValues::NIRFMXSPECAN_INT32_NF_EXTERNAL_PREAMP_PRESENT_TRUE));
```
Comment (gregstoll) [inline source/tests/system/nirfmxspecan_driver_api_tests.cpp:309]: Hmm, I thought we got rid of all the "TRUE" enum values?

Comment (astarche) [reply source/tests/system/nirfmxspecan_driver_api_tests.cpp:309]: Attributes are different. That's a big "attribute_values" enum that includes bool enums as well as non-bool enums. So the simple swap doesn't work.

We could add a `get/set_attribute_bool` wrapper that calls into `get/set_attribute_i32`. I think that would be cool, although nothing would stop you from calling it for some other int32 attribute that's not a bool (though is that worse than allowing passing in cross-enums?).

Comment (gregstoll) [reply source/tests/system/nirfmxspecan_driver_api_tests.cpp:309]: Yeah, the bool wrapper does sound kinda nice but I'm not sure it's worth the complication...

**Review by reckenro — APPROVED** [review]

**Pull request merged**